### PR TITLE
[PR] Initial implementation

### LINKED ISSUE #1: Feat: Create `Flutter` Stopwatch Tutorial
One of the "core" features of our **`App`** is **`timers`**, see: [mvp.fly.dev](https://github.com/dwyl/mvp/issues/140) 📱 
At its' core the **`timer`** feature is a stopwatch. ⏱️ 

# Todo

+ [x] Create a **`Flutter`** tutorial similar to: https://github.com/dwyl/phoenix-liveview-stopwatch
+ [x] Take the reader from zero to working App on a ***Mobile Device*** in as much detail as necessary
+ [x] Bonus Level: Persistence e.g: using `drift` https://github.com/dwyl/learn-flutter/issues/70 to have a _history_ of the timers you have captured on the device.

### PR DESCRIPTION
**Pull request by LuchoTurtle**
fixes #1 

This PR should create the initial implementation of the project and a tutorial for users to get up and running with a project with persisting timers and a simple stopwatch on a real device.

It's still a work in progress as the README still has to be written. But the project itself should be working correctly 😄 .

### CODE REVIEW

**Comment by LuchoTurtle** [comment]
This took much longer than I expected, persisting timers within the device was simply working _against me_. Apparently downloading `.gradle` was corrupted on the initial setup and I spent hours looking at a problem that was just a matter of deleting the corrupted files and doing the first building process again.
I wish the stack trace was exposed on the get-go and the messages were explicit :/

Regardless, what's left now is just finishing up the documentation. The project works on both emulator and on my Android and it persists timers between sessions and the stopwatch seems to be working smoothly, at least 😄

**Comment by nelsonic** [comment]
Great news! 🎉 
Sucks about the `.gradle` file. 😞 
Glad you figured it out in the end. 👌

**Comment by LuchoTurtle** [comment]
This PR should now be complete and the project should work properly. 
😄

**Comment by nelsonic** [comment]
Looking forward to going through this one. 🤞

File: lib/database.g.dart [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  final GeneratedDatabase attachedDatabase;
+  final String? _alias;
+  $TimersTable(this.attachedDatabase, [this._alias]);
+  final VerificationMeta _idMeta = const VerificationMeta('id');
+  @override
+  late final GeneratedColumn<int> id = GeneratedColumn<int>(
+      'id', aliasedName, false,
+      type: DriftSqlType.int,
+      requiredDuringInsert: false,
+      defaultConstraints: 'PRIMARY KEY AUTOINCREMENT');
```
Comment (SimonLab) [inline lib/database.g.dart]: I have the following error on this line:
>The argument type 'String' can't be assigned to the parameter type 'void Function(GenerationContext)?'.

If I delete this line the project compiles and run.
Does anyone else has this error?

File: lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,216 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:drift/drift.dart' as drift;
+import 'package:stopwatch_demo/stopwatch.dart';
+
+import 'database.dart' as Db;
```
Comment (SimonLab) [inline lib/main.dart]: I have a warning (non blocker) on the `Db` name:

>The prefix 'Db' isn't a snake_case identifier.
Try changing the prefix to follow the snake_case style.

File: lib/database.g.dart (hunk 1 above)
Comment (LuchoTurtle) [reply lib/database.g.dart]: This file is generated when you run the "build_runner". I think it's documented on the readme how to generate. Have you tried generating it again and running the file?

Comment (SimonLab) [reply lib/database.g.dart]: Ran `flutter pub run build_runner build --delete-conflicting-outputs` and the file is now updated and errors removed.
I needed to add the flag `--delete-conflicting-outputs` to refresh the file.
Should the `database.g.dart` file be in `.gitignore`?

Readme section where the command is described: https://github.com/dwyl/flutter-stopwatch-tutorial/tree/initial-implementation#persisting-timers

> Run flutter pub run build_runner build and you will notice a database.g.dart file was created. To use this file, change the MyDatabase class defined in the database.dart file defined earlier.

File: lib/main.dart (hunk 2 above)
Comment (LuchoTurtle) [reply lib/main.dart]: Where are these warnings popping up? In VS Code? Just for future reference 👌

Comment (SimonLab) [reply lib/main.dart]: Yes VSCode displays them in the `Problems` tab:
image

**Review by nelsonic — APPROVED** [review]
@LuchoTurtle nice work.

image

image

Thanks! 🚀

**Pull request merged**